src/Microsoft.DotNet.Darc/src/Darc/Helpers/UxHelpers.cs

```diff
@@ -248,5 +248,72 @@ public static StreamWriter GetOutputFileStreamOrConsole(string outputFile)
             }
             return outputStream;
         }
+
+        /// <summary>
+        ///     Verify that a branch exists in the specified repo and contains a version details file.
+        ///     If it does not, optionally prompt the user to confirm that they wish to continue.
+        /// </summary>
+        /// <param name="remote">Remote</param>
+        /// <param name="repo">Repository that the branch should be in</param>
+        /// <param name="branch">Branch to check the existence of</param>
+        /// <param name="prompt">Prompt the user to verify that they want to continue</param>
+        /// <returns>True if the branch exists, prompting is not desired, or if the user confirms that they want to continue. False otherwise.</returns>
+        public static async Task<bool> VerifyAndConfirmBranchExistsAsync(IRemote remote, string repo, string branch, bool prompt)
+        {
+            try
+            {
+                var dependencies = await remote.GetDependenciesAsync(repo, branch);
+            }
+            catch (DependencyFileNotFoundException)
+            {
+                Console.WriteLine($"Warning: Could not find an eng/Version.Details.xml at '{repo}@{branch}'. Dependency updates may not happen as expected.");
+                if (prompt)
+                {
+                    return PromptToContinue();
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        ///     Verify that a repository exists, and optionally prompt the user if it does not.
+        /// </summary>
+        /// <param name="remote">Remote</param>
+        /// <param name="repo">Repository to check for</param>
+        /// <param name="prompt">Prompt the user to verify that they want to continue</param>
+        /// <returns>True if the repository exists, prompting is not desired, or if the user confirms that they want to continue. False otherwise.</returns>
+        public static async Task<bool> VerifyAndConfirmRepositoryExistsAsync(IRemote remote, string repo, bool prompt)
+        {
+            if (!(await remote.RepositoryExistsAsync(repo)))
+            {
+                Console.WriteLine($"Warning: Could not locate repository '{repo}'. Dependency updates may not happen as expected.");
+                if (prompt)
+                {
+                    return PromptToContinue();
+                }
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Prompt the user to continue or not
+        /// </summary>
+        /// <returns>True if we should continue, false otherwise.</returns>
+        public static bool PromptToContinue()
+        {
+            char keyChar;
+            do
```

Comment:
[NIT] Maybe add a few tries before we exit regardless of input? Just to keep in line with other UX operations where we assume failure after a certain number of attempts.

Reply:
Good idea.  Done.